pkg/storkctl/migrationschedule.go

```diff
@@ -60,40 +85,64 @@ func newCreateMigrationScheduleCommand(cmdFactory Factory, ioStreams genericclio
+			// Verify the admin cluster pair exists in the admin namespace
+			if c.Flags().Changed("admin-cluster-pair") {
+				_, err := storkops.Instance().GetClusterPair(adminClusterPair, adminNs)
+				if err != nil {
+					util.CheckErr(fmt.Errorf("unable to find the admin cluster pair in the admin namespace"))
+					return
+				}
+			}
+
 			migrationScheduleNs := cmdFactory.GetNamespace()
 			clusterPairObj, err := storkops.Instance().GetClusterPair(clusterPair, migrationScheduleNs)
 			if err != nil {
 				util.CheckErr(fmt.Errorf("unable to find the cluster pair in the given namespace"))
 				return
 			}
 			isStorageOptionsProvided := true
-			//clusterPairObj.Spec.Options map is empty for syncDr and should be non-empty for volume migration in asyncDR
+			// clusterPairObj.Spec.Options map is empty for syncDr and ideally should be non-empty for volume migration in asyncDR
 			if len(clusterPairObj.Spec.Options) == 0 {
 				isStorageOptionsProvided = false
 			}
 
-			// Default value of includeVolumes for asyncDr with StorageOptions provided is true and for sync DR is false
+			// Default value of includeVolumes when StorageOptions are provided is true and else is false
 			if !isStorageOptionsProvided {
-				//covers the scenario where user sets the --exclude-volumes=false value in the command for a use case where clusterPair's storage-status is Not Provided
-				//this is true if the cluster pair is created for a sync-dr use-case or if clusterPair was created without specifying storage options
+				// covers the scenario where user sets the --exclude-volumes=false value in the command for a use-case where clusterPair's storage-status is Not Provided
+				// which is true if the cluster pair is created for a sync-dr use-case or if clusterPair was created without specifying storage options
 				if c.Flags().Changed(excludeVolumesFlag) && includeVolumes {
-					util.CheckErr(fmt.Errorf("--exclude-volumes can only be set to true if it is a sync-dr use case or storage options are not provided in the cluster pair"))
+					util.CheckErr(fmt.Errorf("--exclude-volumes can only be set to true if it is a sync-dr use case or when storage options are not provided in the cluster pair"))
 					return
 				}
 				includeVolumes = false
 			}
 
-			if len(namespaceList) == 0 {
-				util.CheckErr(fmt.Errorf("need to provide atleast one namespace to migrate"))
+			migrationNamespaces := namespaceList
+			if len(namespaceSelectors) != 0 {
+				// update the migrationNamespaces list by fetching namespaces based on provided label selectors
+				migrationNamespaces, err = getMigrationNamespaces(namespaceList, namespaceSelectors)
+				if err != nil {
+					util.CheckErr(fmt.Errorf("unable to get the namespaces based on the provided --namespace-selectors : %v", err))
+					return
+				}
+			}
+
+			// User must provide at-least one namespace to migrate
+			if len(migrationNamespaces) == 0 {
```

Comment:
Should we add this check? The use case with namespace selector is such that we can create the migrationschedule and may keep it running and add the label later. Even migration should not face any problem when no namespace is found using namespaceselector? @pp511 can you add your thought around it?

Reply:
@diptiranjanpx  We have added other validations in this cli too. For eg: If someone specifies --admin-cluster-pair but doens't have this CR created then it fails. Similarly if an invalid RT spec is not specified it fails. On similar line of thought, if no valid namespaces/namespaceSelectors are specified then it fails. So the behavior is consistent.  If there are no valid namespaces specified in the MigrationSchedule(including namespaces and namespaceSelectors), then migration does fail and better we fail early instead of later during migration.

But you make a valid point. Someone might want to create these migrationSchedule in advance before starting the applications and might expect migrations to fail until the apps have been deployed. And expectation might be that migration should work seamlessly once everything is deployed,. Maybe providing a -f option in the cli to force creation of MigrationSchedule to just print a warning message but not failing the CR creation makes sense?  Let's discuss more on this.

Reply:
Makes sense for other validations. May be not valid for the condition if `namespaceselector is provided but no namespaces found`. 
Let's not go with -f option. We can relax the check in another PR if required.

Reply:
OK. @olavangad-px Can you do this modification as @diptiranjanpx is suggesting?

Reply:
Ok @pp511 . Since I have already merged this PR I will raise another PR to remove the validation then.

Reply:
@pp511 I discussed this with Dipti in the office today.
We feel that the validation makes sense for now since the migrations are failing in this particular case.
We can discuss more on this and take a call later.